docs/content/drafts/Tooltip.mdx

```diff
@@ -0,0 +1,80 @@
+---
+componentId: tooltip
+title: Tooltip
+status: Alpha
+---
+
+import data from '../../../src/drafts/Tooltip/Tooltip.docs.json'
+
+The Tooltip component is used to add context to interactive elements on the page.
+
+**_⚠️ Usage warning! ⚠️_**
+
+Tooltips as a UI pattern should be our last resort for conveying information because it is hidden by default and often with zero or little visual indicator of its existence.
+
+Before adding a tooltip, please consider: Is this information essential and necessary? Can the UI be made clearer? Can the information be shown on the page by default? See [Tooltip alternatives](https://primer.style/design/accessibility/tooltip-alternatives) for more accessible alternatives.
+
+Tooltip should only be used on an element that is interactive such as a button or a link.
+
+## Examples
+
+### Default (As a description type)
+
+```jsx live
+<Tooltip text="This change cannot be undone.">
+  <Button>Delete</Button>
+</Tooltip>
+```
+
+### As a label type
+
+```jsx live
+<Tooltip text="Contribution Documentation for 'Primer React'" type="label">
+  <Link href="https://github.com/primer/react/contributor-docs/CONTRIBUTING.md" sx={{ml: 1, color: 'fg.muted'}}>
+    <StyledOcticon icon={BookIcon} sx={{color: 'fg.muted'}} />
+  </Link>
+</Tooltip>
+```
+
+### With direction
+
+```jsx live
+<Box sx={{padding: 5, display: 'flex', gap: '8px'}}>
+  <Tooltip direction="n" text="Supplementary text" type="description">
+    <Button>North</Button>
+  </Tooltip>
+  <Tooltip direction="s" text="Supplementary text" type="description">
+    <Button>South</Button>
+  </Tooltip>
+  <Tooltip direction="e" text="Supplementary text" type="description">
+    <Button>East</Button>
+  </Tooltip>
+  <Tooltip direction="w" text="Supplementary text" type="description">
+    <Button>West</Button>
+  </Tooltip>
+  <Tooltip direction="ne" text="Supplementary text" type="description">
+    <Button>North East</Button>
+  </Tooltip>
+  <Tooltip direction="nw" text="Supplementary text" type="description">
+    <Button>North West</Button>
+  </Tooltip>
+  <Tooltip direction="se" text="Supplementary text" type="description">
+    <Button>Southeast</Button>
+  </Tooltip>
+  <Tooltip direction="sw" text="Supplementary text" type="description">
+    <Button>Southwest</Button>
+  </Tooltip>
+</Box>
+```
+
+### With no delay
+
+```jsx live
+<Tooltip noDelay text="Supplemetary text">
```

Comment:
I have honestly forgotten if we want to keep `noDelay` or remove it 😅 

I personally think Tooltip should be very snappy by default and we should not have this customisable based on a page 🤔

Reply:
I'll tag @langermank because we are a proposal for some visual changes [in the design repo](https://github.com/primer/design/pull/565) and might be good idea to bring it up there if this is something we are going to go forward with it 👀

Reply:
Oh great callout @siddharthkp! I agree Tooltip should be snappy by default, so I would actually remove this prop and remove the delay in the default styles. If there's a delay in Primer View Components I'll make sure to remove it as well when I update the caret styles.

Reply:
Sounds great! I'll update the code, docs and stories accordingly ✨

Reply:
Love to see it!